src/controller/insertcontent.js

```diff
@@ -226,6 +226,12 @@ class Insertion {
 		if ( node.is( 'element' ) ) {
 			this.handleNodes( node.getChildren(), context );
 		}
+		// When disallowed node is a text but text is allowed in current parent it means that our node
+		// contains disallowed attributes and we have to remove them.
+		else if ( node.is( 'text' ) && this.schema.check( { name: '$text', inside: [ this.position.parent ] } ) ) {
```

Comment:
Uh, sorry. You passed the whole parent there. It's fine. I thought that you passed just a name.

Still, I think that `inside: this.position` might work.